src/request/providers/xhr.ts

```diff
@@ -205,6 +205,19 @@ export default function xhr(url: string, options: XhrRequestOptions = {}): Uploa
 	const task = <UploadObservableTask<XhrResponse>>new Task<XhrResponse>((resolve, reject) => {
 		timeoutReject = reject;
 
+		if (options.signal) {
+			options.signal.addEventListener('abort', () => {
```

Comment:
when the request is fulfilled, the event listener should be removed.

Reply:
@ycabon I'm not sure there's a good place in the code to remove this event listener. Also, I'm not sure it's necessary as subsequent `abort` events will have no effect on a fulfilled Task.

Reply:
My concern was more for cases where the signal is reused for multiple calls, would the Task be kept in memory until the signal and controller are GCed.